pkg/cmd/synonyms/import/import_test.go

```diff
@@ -7,13 +7,102 @@ import (
 	"testing"
 
 	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
+	"github.com/google/shlex"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"github.com/algolia/cli/pkg/cmdutil"
 	"github.com/algolia/cli/pkg/httpmock"
+	"github.com/algolia/cli/pkg/iostreams"
 	"github.com/algolia/cli/test"
 )
 
+func TestNewImportCmd(t *testing.T) {
+	file := filepath.Join(t.TempDir(), "synonyms.ndjson")
+	_ = ioutil.WriteFile(file, []byte("{\"objectID\":\"test\", \"type\": \"synonym\", \"synonyms\": [\"test\"]}"), 0600)
+
+	tests := []struct {
+		name      string
+		tty       bool
+		cli       string
+		wantsErr  bool
+		wantsOpts ImportOptions
+	}{
+		{
+			name:     "no file specified",
+			cli:      "index",
+			wantsErr: true,
+		},
+		{
+			name:     "file not found",
+			cli:      "index --file not-found",
+			wantsErr: true,
+		},
+		{
+			name: "file specified",
+			cli:  fmt.Sprintf("index -F %s", file),
+			wantsOpts: ImportOptions{
+				Index:                   "index",
+				ForwardToReplicas:       true,
+				ReplaceExistingSynonyms: false,
+			},
+		},
+		{
+			name: "forward to replicas",
+			cli:  fmt.Sprintf("index -F %s -f=false", file),
+			wantsOpts: ImportOptions{
+				Index:                   "index",
+				ForwardToReplicas:       false,
+				ReplaceExistingSynonyms: false,
+			},
+		},
+		{
+			name: "replace existing synonyms",
+			cli:  fmt.Sprintf("index -F %s -r=true", file),
+			wantsOpts: ImportOptions{
+				Index:                   "index",
+				ForwardToReplicas:       true,
+				ReplaceExistingSynonyms: true,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			io, _, _, _ := iostreams.Test()
+			if tt.tty {
+				io.SetStdinTTY(tt.tty)
+				io.SetStdoutTTY(tt.tty)
+			}
+
+			f := &cmdutil.Factory{
+				IOStreams: io,
+			}
+
+			var opts *ImportOptions
+			cmd := NewImportCmd(f, func(o *ImportOptions) error {
+				opts = o
+				return nil
+			})
+
+			args, err := shlex.Split(tt.cli)
+			require.NoError(t, err)
+			cmd.SetArgs(args)
+			_, err = cmd.ExecuteC()
+			if tt.wantsErr {
+				assert.Error(t, err)
+				return
+			} else {
```

Comment:
Do we need this `else` here since there's a return at line 94?

Reply:
Good catch 👍